### PR TITLE
add express-rate-limit for rate limiting

### PR DESCRIPTION
**Pull request by jseppi**
ref #1092 

Adds simple rate limiting using [express-rate-limit](https://github.com/nfriedly/express-rate-limit) for all requests.

Also:
- refactor router into separate routers for improved understandability/readability:
  - `apiRouter` for all API routes, ie, those that start with `/v0`
  - `mainRouter` for all other routes
- fix linting errors on touched files

To Do:
- [x] ensure `req.ip` is different per user -- ie, that it is not the cloud.gov router IP
  - Update: `req.ip` is indeed the IP of the proxy, but we can use `app.enable('trust proxy')` to get the correct ip in `req.ip`
- [x] add `app.enable('trust proxy')`
- [x] figure out good parameters to use for the rate limiter
- [x] move rate limit options to config

### CODE REVIEW

**Comment by jseppi** [comment]
lol, the rate limiting broke the tests

File: app.js [hunk 1]
```diff
@@ -65,8 +72,17 @@ app.use(expressWinston.errorLogger({
   ],
 }));
 
-const routers = require('./api/routers');
+const limiter = new RateLimit({
```
Comment (jmhooper) [inline app.js]: One thought I have is we could add this to a config, something like `config.rateLimitting`. Then you could configure production like this, and do this for the default:

```
{
  max: 0,
  delayAfter: 0,
  delayMs: 0
}
```

That should enable rate limiting in prod, but disable it in dev and test. But we could still get and idea of how the app behaves when requests are passed through the middleware in the test and dev environment.

Thoughts?

**Comment by jseppi** [comment]
Yeah, that's a good idea!

**Comment by jseppi** [comment]
ready for review again, @jmhooper

**Pull request merged**